Add ipu-plugin and sriov_cni code (#1)
intel/ipu-opi-plugins: docs/deployment.md

drwxr-xr-x 2 root root    4096 Oct  4  2023 runtime_files
-rw-r--r-- 1 root root 1442232 Oct  4  2023 e2100-default-1.0.12.0.pkg
```
**_NOTE:_** the symbolic link from default_pkg.pkg to /etc/dpcp/package/linux_networking.pkg. \
If this symbolic link is missing then likely the following went wrong: \
(1) the script to load the custom package may contain some error, or\
(2) the script to load the custom package was not executed during startup of the IMC.

### Enable Communication
On ACC, we should see something like this:
```bash
[    6.360336] idpf 0000:00:01.0 enp0s1f0d1: renamed from eth1  
[    6.569522] idpf 0000:00:01.0 enp0s1f0d2: renamed from eth2 <-- USED TO COMMUNICATE WITH HOST
[    6.679568] idpf 0000:00:01.0 enp0s1f0d6: renamed from eth6
[    6.909360] idpf 0000:00:01.0 enp0s1f0d4: renamed from eth4
[    7.039378] idpf 0000:00:01.0 enp0s1f0d5: renamed from eth5  
[    7.159385] idpf 0000:00:01.0 enp0s1f0d3: renamed from eth3 
[    7.299356] idpf 0000:00:01.0 enp0s1f0: renamed from eth0   <-- USED TO COMMUNICATE WITH IMC
[    7.459362] idpf 0000:00:01.0 enp0s1f0d7: renamed from eth7
```

> On ACC, enp0s1f0 (renamed from eth0) and enp0s1f0d2 (renamed from eth2) can be used to set up the communication with IMC and HOST.

Reviewer: ```suggestion
On ACC:

- enp0s1f0 (renamed from eth0) will be used to set up the communication with IMC.

- enp0s1f0d2 (renamed from eth2) will be used to set up the communication with HOST.

```

                "context": "/root/linux_networking/linux_networking.context.json",
                "config": "/root/linux_networking/tofino.bin",
                "pipe_scope": [
                    0,
                    1,
                    2,
                    3
                ],
                "path": "/root/linux_networking"
            }
            ]
        }
        ],
        "agent0": "lib/libpltfm_mgr.so"
    }
    ]
}
```

#### 3.4 Generate the Binary File

On ACC, use `tdi_pipeline_builder` available under `/opt/p4/p4-cp-nws/bin` to
combine the artifacts generated by the `p4c-pna-xxp` compiler and generate
forwarding pipeline binary.

```bash
$P4CP_INSTALL/bin/tdi_pipeline_builder \
    --p4c_conf_file=/usr/share/stratum/es2k/es2k_skip_p4.conf \
    --bf_pipeline_config_binary_file=/root/linux_networking/linux_networking.pb.bin
```

### Set Huge Pages

On ACC, run the following command:

```bash
mkdir /dev/hugepages
mount -t hugetlbfs -o pagesize=2M none /dev/hugepages
echo 512 > /sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages
```

###  Start Infrap4d Process <a name="start_infrap4d_process"></a>

On ACC, run the command below:

```bash

> /opt/p4/p4-cp-nws/sbin/infrap4d -grpc_open_insecure_mode=true -nodetach -disable_krnlmon=true

Reviewer: ```suggestion
/opt/p4/p4-cp-nws/sbin/infrap4d -grpc_open_insecure_mode=true -nodetach 
```

```
 
The following options are used:

- `-grpc_open_insecure_mode=true` is used because the setup described in this document uses an insecure communication mode. 

- `-nodetach` is used to avoid running the `infrap4d` in a detached mode. This is not a strong requirement, but it is useful to see the log generated by the `infrap4d` process. Note that another terminal needs to be used to proceed as the current one will be used by the `infrap4d` process. In case the detached mode is used, double check that the `infrap4d` process has been started and it is running.

> - `-disable_krnlmon=true` is used to run the `infrap4d` without krnlmon support. This option can be omitted, but it has been used to make log generated by the binary for setting the P4 rules easily visible. 

Reviewer: Remove the paragraph about `-disable_krnlmon=true`

See `/opt/p4/p4-cp-nws/sbin/infrap4d --help` for more options.

> **Note**: In case that the `infrap4d` fails to start, try to run the start command in the folder where the `linux_networking` recipe files are stored e.g., `cd /root/linux_networking`. Sometimes this command fails and needs to be re-run. 

Reviewer: This note should be removed

```bash
$P4CP_INSTALL/bin/p4rt-ctl set-pipe br0 /root/linux_networking/linux_networking.pb.bin /root/linux_networking/linux_networking.p4info.txt
```

###  Configure the VSI Group and Add a netdev <a name="configure_vsi_group"></a>

Use one of the IPDF netdevs on ACC to receive all control packets from overlay VM's by assigning to a VSI group. 

On ACC, run the command below to select an interface.

```bash
[root@mev-acc-rl linux_networking]# ip -br l
...
enp0s1f0         UP     00:00:00:00:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d1       UP     00:08:00:01:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d2       UP     00:09:00:02:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d3       UP     00:0a:00:03:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>  <-- WE USE THIS
enp0s1f0d4       UP     00:0b:00:04:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d5       UP     00:0c:00:05:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d6       UP     00:0d:00:06:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
enp0s1f0d7       UP     00:0e:00:07:03:18 <BROADCAST,MULTICAST,UP,LOWER_UP>
...
```
Warning: Do not reuse same interface used for ACC and HOST communication

> In this document, the `enp0s1f0d3` is selected (the dots are added to show that other network interfaces may already be available).

Reviewer: ```suggestion
In this document, `enp0s1f0d3` is selected (the dots are added to show that other network interfaces may already be available).
```
